addons/crm/models/crm_lead.py

```diff
@@ -286,7 +286,7 @@ def _compute_company_id(self):
                 if lead.team_id and not lead.team_id.company_id and not lead.user_id:
                     proposal = False
                 # no user and no team -> void company and let assignment do its job
-                if not lead.team_id and not lead.user_id:
+                if not lead.team_id and not lead.user_id and not (lead.partner_id and proposal == lead.partner_id.company_id):
```

Comment:
@tfr-odoo Makes sense to me, but as you have seen things we cannot imagine, what do you think of that PR ?

Reply:
I didn't see things like that because we have remove the check company auto (which maybe is another approach to solve this issue) and partner have no company set